master/internal/job/job.go

```diff
@@ -133,6 +141,206 @@ func (j *Jobs) jobQSnapshot(ctx *actor.Context, resourcePool string) (AQueue, er
 	return jobQ, nil
 }
 
+// figure out what changes are needed to move the job to the desired position.
+// Later we can support cross RP moving this way as well.
+func moveJobMessages(
+	jobs []*jobv1.Job,
+	target *jobv1.Job,
+	anchor *jobv1.Job,
+	anchorIdx int,
+	aheadOf bool,
+) (*SetGroupPriority, *MoveJob, error) {
+	// validate anchorIdx
+	if anchorIdx < 0 || anchorIdx > len(jobs) {
+		return nil, nil, fmt.Errorf("invalid anchor index %d", anchorIdx)
+	}
+	// validate anchor and target
+	if anchor == nil {
+		return nil, nil, fmt.Errorf("missing anchor job")
+	}
+	if target == nil {
+		return nil, nil, fmt.Errorf("missing target job")
+	}
+	if target.JobId == anchor.JobId {
+		return nil, nil, fmt.Errorf("target and anchor jobs are the same")
+	}
+
+	// sanity check
+	lastJob := int32(-1)
+	for _, job := range jobs {
+		check.Panic(check.GreaterThanOrEqualTo(job.Summary.JobsAhead, lastJob))
+		lastJob = job.Summary.JobsAhead
+	}
+
+	var priorityMsg *SetGroupPriority
+	if target.Priority != anchor.Priority {
+		priorityMsg = &SetGroupPriority{
+			Priority: int(anchor.Priority),
+		}
+	}
+
+	var moveMsg *MoveJob
+	// find the next or previous job based on aheadOf in the same priority lane
+	var secondAnchor model.JobID
+	if aheadOf {
+		for idx := anchorIdx - 1; idx >= 0; idx-- {
+			if jobs[idx].Priority == anchor.Priority {
```

Comment:
can you clarify how this would work? What if there are multiple jobs within the priority? Would we just pick the last one/first one in the range and use that as the secondAnchor? Would this be a problem?

Reply:
the idea is to pick the first one right after, or right behind, in the same priority lane, and if there are none we use head or tail anchors to avoid making those edge cases.

Reply:
I might be missing cases where this fails. let me know if you see some

Reply:
So the specific question I'm wondering here is why we're searching for the next priority level instead of searching for the job right before or right after the anchor? 

an example of what I'm concerned about is if there are more than 2 jobs within a priority and you're trying to move within that priority. So if you have job A, which is priority 1, and jobs B, C, D, E and F which are priority 2. What happens if you try to promote job F above job D?